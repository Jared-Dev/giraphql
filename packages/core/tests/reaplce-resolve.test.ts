import { execute, lexicographicSortSchema, parse, printSchema } from 'graphql';
import pluginSchema from './examples/replace-resolve';

describe('replace-resolve plugin example', () => {
  it('generates expected schema', () => {
    expect(printSchema(lexicographicSortSchema(pluginSchema))).toMatchSnapshot();
  });

  it('resolves hello', async () => {
    const result = await execute({
      schema: pluginSchema,
      document: parse(/* GraphQL */ `
        query {
          hello
        }
      `),
    });

    expect(result).toEqual({
      data: {
        hello: 'world',
      },
    });
  });
});
