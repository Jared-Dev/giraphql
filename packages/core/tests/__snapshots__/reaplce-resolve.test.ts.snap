// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`replace-resolve plugin example > generates expected schema 1`] = `
"type Query {
  hello: String
}"
`;
